mythril/laser/ethereum/state.py

```diff
@@ -1,16 +1,44 @@
-from z3 import BitVec, BitVecVal, Solver, ExprRef, sat
+from z3 import BitVec, BitVecVal, Solver, ExprRef, Concat, sat, simplify
 from mythril.disassembler.disassembly import Disassembly
 from copy import copy, deepcopy
 from enum import Enum
 from random import randint
 
 from mythril.laser.ethereum.evm_exceptions import StackOverflowException, StackUnderflowException
 
-
 class CalldataType(Enum):
     CONCRETE = 1
     SYMBOLIC = 2
 
+class SymbolicCalldata:
+    def __init__(self, tx_id: int, starting_calldata: bytes=None):
+        self.tx_id = tx_id
+        self._calldata = {}
+
+        if starting_calldata:
+            for i in range(len(starting_calldata)):
+                self._calldata[i] = BitVecVal(starting_calldata[i], 8)
+
+    def get_word_at(self, index: int):
+        return self[index:index+32]
+
+    def __getitem__(self, item: int):
+        if isinstance(item, slice):
+            if item.step != None \
+            or item.start > item.stop \
+            or item.start < 0 \
+            or item.stop < 0: raise IndexError("Invalid Calldata Slice")
+
+            dataparts = []
+            for i in range(item.start, item.stop):
+                dataparts.append(self[i])
```

Comment:
I think we can compress the above 3 lines to something like `dataparts = [self[i] for i in range(item.start, item.stop)]`